tests/healthcare/healthcare_integration_test.go

```diff
@@ -0,0 +1,613 @@
+			api:           "http://127.0.0.1:5000/api/tool/my-list-dicom-stores-tool/invoke",
+			requestHeader: map[string]string{"Authorization": accessToken},
+			requestBody:   bytes.NewBuffer([]byte(`{}`)),
+			want:          want,
+			isErr:         false,
+		},
+		{
+			name:          "invoke my-auth-list-dicom-stores-tool without auth token",
+			api:           "http://127.0.0.1:5000/api/tool/my-auth-list-dicom-stores-tool/invoke",
+			requestHeader: map[string]string{},
+			requestBody:   bytes.NewBuffer([]byte(`{}`)),
+			isErr:         true,
+		},
+		{
+			name:          "invoke my-auth-list-dicom-stores-tool with invalid auth token",
+			api:           "http://127.0.0.1:5000/api/tool/my-auth-list-dicom-stores-tool/invoke",
+			requestHeader: map[string]string{"Authorization": "invalid-token"},
+			requestBody:   bytes.NewBuffer([]byte(`{}`)),
+			isErr:         true,
+		},
+	}
+	for _, tc := range invokeTcs {
+		t.Run(tc.name, func(t *testing.T) {
+			runTest(t, tc.api, tc.requestHeader, tc.requestBody, tc.want, tc.isErr)
+		})
+	}
+}
+
+func runTest(t *testing.T, api string, requestHeader map[string]string, requestBody io.Reader, want string, isErr bool) {
+	req, err := http.NewRequest(http.MethodPost, api, requestBody)
+	if err != nil {
+		t.Fatalf("unable to create request: %s", err)
+	}
+	req.Header.Add("Content-type", "application/json")
+	for k, v := range requestHeader {
+		req.Header.Add(k, v)
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatalf("unable to send request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		if isErr {
+			return
+		}
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		t.Fatalf("response status code is not 200, got %d: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	var body map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&body)
+	if err != nil {
+		t.Fatalf("error parsing response body")
+	}
+
+	got, ok := body["result"].(string)
+	if !ok {
+		t.Fatalf("unable to find result in response body")
+	}
+
+	if !strings.Contains(got, want) {
+		t.Fatalf("expected %q to contain %q, but it did not", got, want)
+	}
+}
+
+func runListFHIRStoresWithRestriction(t *testing.T, allowedFHIRStore, disallowedFHIRStore string) {
+	api := "http://127.0.0.1:5000/api/tool/list-fhir-stores-restricted/invoke"
+	req, err := http.NewRequest(http.MethodPost, api, bytes.NewBuffer([]byte(`{}`)))
+	if err != nil {
+		t.Fatalf("unable to create request: %s", err)
+	}
+	req.Header.Add("Content-type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatalf("unable to send request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		t.Fatalf("response status code is not 200, got %d: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	var body map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&body)
```

Comment:
```suggestion
	err = json.Unmarshal(bodyBytes ,&body)
```

Could be updated to Unmarshal after using `tests.RunRequest()`